### DISCUSSION TITLE
How should we implement the logger?

### DISCUSSION THREAD

**Discussion by mapsgeek** [ai/nanostores#191] [Q&A] [answered]
how to use [logger ](https://github.com/nanostores/logger) as devtools?

**Comment by ai** [comment]
Add logger according to instruction and open log in the standard DevTools of your browser

**Reply by euaaaio** [reply]
We still haven't done an instruction. But I'm working on it — I'll release an update this month.

**Comment by euaaaio** [comment, marked as answer]
Here is a simple example:

```js
import { logger } from '@nanostores/logger'

logger({
  UserStore,
  PostsStore
})
```

**Reply by euaaaio** [reply]
Also we have an option for `mapTemplate` (deprecated, but part of Logux Client) to name different stores in logs:

```js
logger(
  {
    …
  },
  {
    nameGetter: (store: AnyStore, templateName: string) => string
  }
)
```

Default name getter: `${templateName}:${store.get().id}`

**Comment by mapsgeek** [comment]
Awesome, thank you

**Comment by neodon** [comment]
Hey I just wanted to say I love the work you're doing with the logger. I managed to hack it into my codebase today and it helped me solve a mess I had made. It's already useful so I'm eager to see where it goes and when I can add it to the dependencies for my project.

**Reply by euaaaio** [reply]
Thanks 🫶🏻
I hope to release an update with support for new versions of Nano Stores this week.

**Comment by euaaaio** [comment]
Nano Stores Logger v0.1 released.
Here is the new documentation: https://github.com/nanostores/logger/blob/main/README.md
